superset-frontend/src/SqlLab/actions/sqlLab.js

```diff
@@ -136,8 +139,23 @@ export function getUpToDateQuery(rootState, queryEditor, key) {
   };
 }
 
-export function resetState() {
-  return { type: RESET_STATE };
+export function resetState(data) {
+  return (dispatch, getState) => {
+    const { common, user, config } = getState();
+    const initialState = getInitialState({
```

Comment:
Why do we send all parameters (`common`, `user`, `config`, etc) when resetting but we don't send them when initializing (`App`, `store`)?

Reply:
good point. I don't like to pass these but sqllab initial logic requires `common`. let me pass this value (common) only